File: C# for VB/3-csharp-visual-basic-dotnet-developers-m3-exercise-files/demos/cs/CSObjects/Square.cs
using System;

namespace CSObjects
{
    public class Square : ShapeBase, 
                          IIdentifiable, 
                          IIdentifiable3,
                          IIdentifiable4
    {
        public Square() : this(new DefaultRenderingProvider()) { }

        public Square(IRenderingProvider provider) : base(provider) { }

        public override void Draw()
        {
            Console.WriteLine("[]");
            base.Draw();
        }

        public override void Fill(string colorName)
        {
            Console.WriteLine("Filling square: " + colorName);
        }

        new public string Identify()
        {
            return "Identify [Square:IIdentifiable]";
        }

        string IIdentifiable3.Identify()
        {
            return "Identify [Square:IIdentifiable3]";
        }

        void IIdentifiable4.Identify()
        {
            Console.WriteLine("Identify [Square:IIdentifiable4]");
        }
    }
}

File: C# for VB/3-csharp-visual-basic-dotnet-developers-m3-exercise-files/demos/vb/VBObjects/Square.vb
Public Class Square
    Inherits ShapeBase

    Public Sub New()
        Console.WriteLine("Constructor")
    End Sub

    Public Sub New(provider As IRenderingProvider)
        MyBase.New(provider)
    End Sub

    Public Overrides Function Identify() As String
        Return "Identify [Square]"
    End Function

    Public Overrides Sub Draw()
        Console.WriteLine("[]")
    End Sub

    Public Overrides Sub Fill(colorName As String)
        Console.WriteLine("Filling square: " & colorName)
    End Sub

End Class

